drivers/clock_control/clock_control_rts5912_sccon.c

```diff
@@ -0,0 +1,242 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2024 Realtek Semiconductor Corporation, SIBG-SD7
+ * Author: Lin Yu-Cheng <lin_yu_cheng@realtek.com>
+ */
+
+#define DT_DRV_COMPAT realtek_rts5912_sccon
+
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/logging/log.h>
+
+#include <zephyr/drivers/clock_control/clock_control_rts5912.h>
+#include <zephyr/dt-bindings/clock/rts5912_clock.h>
+
+#include <reg/reg_system.h>
+
+LOG_MODULE_REGISTER(sccon, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+#define RC25M_FREQ (25000000UL)
+#define PLL_FREQ   (100000000UL)
+
+struct rts5912_sccon_config {
+	uint32_t reg_base;
+	uint8_t sysclk_src;
+	uint8_t sysclk_div;
+};
+
+static int rts5912_periph_clock_control(const struct device *dev, clock_control_subsys_t sub_system,
+					bool on_off)
+{
+	const struct rts5912_sccon_config *const config = dev->config;
+	struct rts5912_sccon_subsys *subsys = (struct rts5912_sccon_subsys *)sub_system;
+
+	SYSTEM_Type *sys_reg = (SYSTEM_Type *)config->reg_base;
+
+	uint32_t clk_grp = subsys->clk_grp;
+	uint32_t clk_idx = subsys->clk_idx;
+
+	uint32_t module_idx;
+
+	switch (clk_grp) {
+	case RTS5912_SCCON_I2C:
+		module_idx = ((clk_idx - I2C0_CLKPWR) >> 2);
+
+		if (on_off) {
+			sys_reg->I2CCLK |= (0x01ul << clk_idx);
```

Comment:
Use the `BIT()` macro in all places where you're using `clk_idx`.
```suggestion
			sys_reg->I2CCLK |= BIT(clk_idx);
```

Reply:
Thanks for recommend, our base will use this way to improve readability.